### PR TITLE
=act #15914 Clarify and correct max-received-message-size

### PR DESCRIPTION
**Pull request by patriknw**
Refs #15914

### CODE REVIEW

File: akka-actor/src/main/resources/reference.conf [hunk 1]
```diff
@@ -557,6 +557,9 @@ akka {
       # The maximum number of bytes delivered by a `Received` message. Before
       # more data is read from the network the connection actor will try to
       # do other work.
+      # The purpose of this setting is to impose a smaller limit than the 
+      # configured receive buffer size. When using value 'unlimited' it will
+      # try to read all from the receive buffer.
       max-received-message-size = unlimited
```
Comment (drewhk) [inline akka-actor/src/main/resources/reference.conf]: should this remain unlimited by default?

**Comment by 2m** [comment]
LGTM

**Comment by ktoso** [comment]
LGTM

File: akka-actor/src/main/resources/reference.conf (hunk 1 above)
Comment (rkuhn) [inline akka-actor/src/main/resources/reference.conf]: yes, I’d think so: while this is `unlimited` the true limit is controlled in exactly one place—preferable to having two places to modify

**Comment by rkuhn** [comment]
LGTM

**Pull request merged**